app/src/main/java/org/oppia/android/app/onboarding/OnboardingViewModel.kt

```diff
@@ -9,8 +9,14 @@ import javax.inject.Inject
 class OnboardingViewModel @Inject constructor() : ObservableViewModel() {
   val slideNumber = ObservableField<Int>(0)
   val totalNumberOfSlides = TOTAL_NUMBER_OF_SLIDES
+  val slideDotsContentDescription = ObservableField<String>("Slide 1 of 4")
 
   fun slideChanged(slideIndex: Int) {
     slideNumber.set(slideIndex)
+    setSlideDotsContentDescription(slideIndex)
+  }
+
+  private fun setSlideDotsContentDescription(slideIndex: Int) {
+    slideDotsContentDescription.set("Slide ${slideIndex + 1} of 4")
```

Comment:
Do not hard code this text.
All strings should be part of `strings.xml` file.

The best way to handle this is by sending the `slideIndex / slideNumber` in the xml file.
And in xml file you can use it similar to what we have done for `chapter_name` string.

If you apply this suggestion mostly you won't need any changes in this file.

Reply:
Thanks for guiding me here. I have modified the approach accordingly. PTAL.